.github/workflows/canary-integration-test.yml

```
@@ -823,3 +814,83 @@ jobs:
+
+    - name: install deps
+      run: |
+        tests/scripts/github-action-helper.sh install_deps
+        sudo apt-get install -y s3cmd
+
+    - name: setup minikube
+      uses: manusa/actions-setup-minikube@v2.3.1
+      with:
+        minikube version: 'v1.18.1'
+        kubernetes version: 'v1.19.2'
+        start args: --memory 6g --cpus=2
+        github token: ${{ secrets.GITHUB_TOKEN }}
+
+    - name: use local disk into two partitions
+      run: |
+        BLOCK=$(sudo lsblk --paths|awk '/14G/ {print $1}'| head -1)
+        tests/scripts/github-action-helper.sh use_local_disk
+        tests/scripts/create-bluestore-partitions.sh --disk "$BLOCK" --osd-count 2
+        sudo lsblk
+
+    - name: build rook
+      run: tests/scripts/github-action-helper.sh build_rook
+
+    - name: deploy first cluster rook
+      run: |
+        tests/scripts/github-action-helper.sh deploy_first_rook_cluster
+        kubectl create -f cluster/examples/kubernetes/ceph/object-multisite-test.yaml
+        # 200s necessary wait for multisite store to be created
+        sleep 200
+
+    - name: prep second cluster pull realm config
+      run: |
+        cd cluster/examples/kubernetes/ceph/
+        IP_ADDR=$(kubectl -n rook-ceph get svc rook-ceph-rgw-multisite-store -o jsonpath="{.spec.clusterIP}")
+        sed -i 's/10.103.133.16/'"$IP_ADDR"'/g' object-multisite-pull-realm-test.yaml
+        yq w -i -d1 object-multisite-pull-realm-test.yaml spec.pull.endpoint http://${IP_ADDR}:80
+        BASE64_ACCESS_KEY=$(kubectl -n rook-ceph get secrets realm-a-keys -o jsonpath="{.data.access-key}")
+        BASE64_SECRET_KEY=$(kubectl -n rook-ceph get secrets realm-a-keys -o jsonpath="{.data.secret-key}")
+        sed -i 's/VzFjNFltMVdWRTFJWWxZelZWQT0=/'"$BASE64_ACCESS_KEY"'/g' object-multisite-pull-realm-test.yaml
+        sed -i 's/WVY1MFIxeExkbG84U3pKdlRseEZXVGR3T3k1U1dUSS9KaTFoUVE9PQ==/'"$BASE64_SECRET_KEY"'/g' object-multisite-pull-realm-test.yaml
+
+    - name: deploy second cluster rook
+      run: |
+        tests/scripts/github-action-helper.sh deploy_second_rook_cluster
+        kubectl create -f cluster/examples/kubernetes/ceph/object-multisite-pull-realm-test.yaml
+        # 200s necessary wait for realms to be pulled and zone-b-multisite-store to be created
+        sleep 200
+
+    - name: wait for ceph cluster 1 to be ready
+      run: |
+        mkdir test
+        tests/scripts/validate_cluster.sh osd 1
+        kubectl -n rook-ceph get pods
+
+    - name: write an object to one cluster, read from the other
+      run: tests/scripts/github-action-helper.sh write_object_to_cluster1_read_from_cluster2
+
+    - name: upload test result
+      uses: actions/upload-artifact@v2
+      if: always()
+      with:
+        name: rgw-multisite-testing
+        path: test
+
+    - name: setup tmate session for debugging
+      if: failure() && contains(github.event.pull_request.labels.*.name, 'rgw-multisite-testing')
+      uses: mxschmitt/action-tmate@v3
```

```suggestion
    - name: setup tmate session for debugging
      if: failure()
      uses: mxschmitt/action-tmate@v3
      timeout-minutes: 120
```
similar to what we have in all other places, recently we enabled tmate for all failure with 120min.